welearn_datastack/modules/sdgs_classifiers.py

```diff
@@ -81,14 +87,20 @@ def n_classify_slice(_slice: DocumentSlice, classifier_model_name: str) -> Sdg |
     embedding: numpy.ndarray = numpy.frombuffer(
         bytes(binary_slice_emb), dtype=numpy.float32
     )
-    tmp_ds_sdg = classifier_model.predict(embedding.reshape(1, -1))
+    tmp_ds_sdg = classifier_model.predict_proba(embedding.reshape(1, -1))
     ds_sdg = tmp_ds_sdg[0]
-    logger.debug("Slice classified as SDG: %s", ds_sdg)
-    if ds_sdg.sum() == 1:
-        ret = ds_sdg.argmax() + 1
-        try:
-            ret = int(ret)
-            return Sdg(slice_id=_slice.id, sdg_number=ret, id=uuid.uuid4())
-        except ValueError:
-            logger.error("SDG is not an integer: %s", ret)
+
+    # Prepare probability list for score sort
+    proba_lst: list[tuple[int, float]] = [
+        (i + 1, proba) for i, proba in enumerate(ds_sdg) if i + 1 in forced_sdg
```

Comment:
we could filter out the lower scores here
```suggestion
        (i + 1, proba) for i, proba in enumerate(ds_sdg) if i + 1 in forced_sdg and proba > 0.5
```